1_fetch/src/fetch_sites.R

```diff
@@ -29,12 +36,82 @@ fetch_sites_from_sf <- function(ind_file, sf_ind, dates, stream_params) {
     # if a gage goes out during the storm, this filter would still capture that gage
     filter(end_date >= as.Date(dates$start))
 
-  # return only the site numbers
-  sites <- sites_df$site_no
+  sites <- dplyr::select(sites_df, site_no, station_nm, dec_lat_va, dec_long_va)
 
   # write the data file and the indicator file
   data_file <- as_data_file(ind_file)
   saveRDS(sites, data_file)
   gd_put(ind_file, data_file)
+}
 
+#' Get list of corresponding NWS site numbers from NWIS site numbers
+#'
+#' @param state_cds vector of states for this storm
+fetch_nws_to_nwis_crosswalk <- function(state_cds){
+
+  conversion_table <- data.frame()
+  for(s in state_cds) {
+    conversion_url <- sprintf("http://www.nws.noaa.gov/oh/hads/USGS/%s_USGS-HADS_SITES.txt", s)
+    conversion_table_state <- readr::read_delim(conversion_url,
+                                                delim = "|",skip = 4,col_names = FALSE,
+                                                col_types = readr::cols())
+    conversion_table <- dplyr::bind_rows(conversion_table, conversion_table_state)
+  }
+
+
+  names(conversion_table) <- c("NWS","USGS","GOES","NWS HSA","lat","lon","name")
+  conversion_table$USGS <- gsub(" ","", conversion_table$USGS)
+  nws_nwis_crosswalk <- dplyr::select(conversion_table, site_no_nws = NWS, site_no=USGS)
+
+  return(nws_nwis_crosswalk)
+}
+
+#' Download National Weather Service data for sites and filter any without stage data
+#'
+#' @param ind_file character file name where the output should be saved
+#' @param nwis_sites_ind indicator file for data.frame with col `site_no` to consider for this storm
+#' @param nws_nwis_crosswalk data.frame with NWS site numbers and their corresponding USGS site number
+fetch_nws_data <- function(ind_file, nwis_sites_ind, nws_nwis_crosswalk) {
+
+  nwis_sites <- readRDS(sc_retrieve(nwis_sites_ind))
+  sites_df <- left_join(nwis_sites, nws_nwis_crosswalk, by="site_no")
+
+  # Start by setting stage info as NA
+  sites_df$flood_stage <- NA
+  sites_df$flood_stage_units <- NA
+
+  # Download NWS info for each gage
+  site_nums_to_query <- unique(sites_df$site_no_nws)
+
+  i_count <- 0
+  for(i in site_nums_to_query){
+
+    if(is.na(i)) {
+      i_count <- i_count + 1
+      print(paste(i_count, "/", length(site_nums_to_query)))
+      next
+    }
+
+    url_site <- paste0("https://water.weather.gov/ahps2/hydrograph_to_xml.php?gage=",i,"&output=xml")
```

Comment:
oof, i see why this loop takes so long! i think that's ok, though, because we should really only have to do this once per storm (unless we need to update our viewbox repeatedly...hmm...). and also, i don't know of a faster way to do it, so there we are.

Reply:
I think there is a waterwatch service or web table we could potentially scrape for this...?

Reply:
Happy to see this get addressed in a future PR if such service/table does exist